test/test_io.py

```diff
@@ -19,12 +19,44 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
+from contextlib import contextmanager
 import io
 import unittest
 import tempfile
+import fastavro
 from lsst.alert.packet.io import retrieve_alerts
+from lsst.alert.packet.schema import Schema
+
 
 class RetrieveAlertsTestCase(unittest.TestCase):
+    def setUp(self):
+        # We could use the real schemas, but that would require building
+        # complicated mocks since there are lots of non-null fields in the real
+        # schema without defaults, and we'd need to test across versions.
+        #
+        # Writing our own schema keeps tests simple, and it also checks that
+        # retrieve_alerts is independent of any of our schemas' versioning or
+        # details.
+        self.test_schema_dict = {
+            "type": "record",
+            "name": "test_record",
+            "fields": [
+                {"name": "id", "type": "long"},
+            ]
+        }
+        self.test_schema = Schema(self.test_schema_dict)
+
+    def _mock_records(self, n):
+        """ Return a list of records with mock values, matching self.test_schema. """
```

Comment:
Closing `"""` [should be on a new line](https://developer.lsst.io/python/numpydoc.html#docstrings-should-begin-with-and-terminate-with-on-its-own-line). And I don't think we want the initial whitespace.

Reply:
I really have to fix up my python-mode docstring style, don't I. Fixed.